metroae

```diff
@@ -155,6 +167,130 @@ function check_run_mode {
     debug "${FUNCNAME[0]}: RUN_MODE is $RUN_MODE"
 }
 
+function phone_home {
+    if [[ $UUID == "" ]]; then
+        debug "${FUNCNAME[0]}: Skipping Phone home"
+    else
+        # append and create json data
+        json_data=""
+        while [[ $# -gt 0 ]]; do
+          json_data+="$1,"
+          shift
+        done
+        json_data=`echo $json_data | sed 's/.$//'`
+        debug "${FUNCNAME[0]}: curl -vvvv -k -H \"Content-Type: application/json\"\
+          -H \"x-apikey: $PHONE_HOME_API_KEY\"\
+          -X POST -d '{$json_data}'\
+          $PHONE_HOME_URL"
+
+        curl -vvvv -k -H "Content-Type: application/json"\
+          -H "x-apikey: $PHONE_HOME_API_KEY"\
+          -X POST -d "{$json_data}"\
+          $PHONE_HOME_URL >> /dev/null 2>&1 &
+    fi
+}
+
+function  check_if_internal_user {
+    debug ${FUNCNAME[0]}
+
+    if [[ $SKIP_UUID -eq $TRUE ]]; then
+        debug "${FUNCNAME[0]}: Using existing UUID $UUID"
+    else
+        int_user_check=`ping -c 5 github.mv.usa.alcatel.com 2>&1`
+        debug "${FUNCNAME[0]} $int_user_check"
+        SKIP_UUID=$?
+
+        if [[ $SKIP_UUID -eq $FALSE ]]; then
+            echo ""
+            write_to_screen_and_script_log "We have detected that this setup of MetroAE is internal to Nokia."
+            echo ""
+
+            declare -l continue_confirm
+            continue_confirm="init"
+            while [[ $continue_confirm != "y" ]] && [[ $continue_confirm != "n" ]] && [[ $continue_confirm != "" ]]
+            do
+                read -p " Would like to share usage data with MetroAE team? (y/N): " continue_confirm
+            done
+
+            if [[ $continue_confirm != "y" ]] ; then
+                SKIP_UUID=$TRUE
+            fi
+        fi
+    fi
+}
+
+function generate_UUID {
+    debug ${FUNCNAME[0]}
+    UUID=`uuidgen`
+    debug "${FUNCNAME[0]}: UUID=$UUID"
+}
+
+function generate_and_save_UUID {
+    debug ${FUNCNAME[0]}
+    if [[ $SKIP_UUID -eq $TRUE ]]; then
+        debug "${FUNCNAME[0]}: Skipping UUID generation"
+    else
+
+        check_if_internal_user
```

Comment:
This is going to incur a time penalty for waiting for the ping to timeout.  For containers, this is called from container_setup which is fine.  But, from repo, this is called from deployment_main.  Will a user have to wait for timeout every single time they issue a command?

Reply:
No they will not be paying any time penalty because the check write above it will skip this section all together. The only it will prompted is the first time the user run it. Once we decide that the user is not internal or if the user opted out we put the SKIP_UUID into .uuid file and source at the beginning and keep skipping this step.

Reply:
Is there a use case where the user gives us permission, we start sending things home, then something changes in their environment and we can no longer reach the network? Will this user be paying a time penalty in such a case?

Reply:
No, once they opt in or out the SKIP_UUID flag is set to skip.